src/renderer/components/Player.tsx

```diff
@@ -67,9 +69,13 @@ export default class Player extends React.Component {
               maxToRememberInHistory={500}
               timingFunction={this.props.scene.timingFunction}
               zoomType={this.props.scene.zoomType}
+              zoomLevel={this.props.scene.zoomLevel}
               isPlaying={this.state.isPlaying}
               fadeEnabled={this.props.scene.crossFade}
               allPaths={this.state.allPaths} />)}
+          {this.state.isLoaded && this.props.scene.hastebinID && this.state.isPlaying && (
+            <CaptionProgram hastebinID={this.props.scene.hastebinID} />
+          )}
```

Comment:
The best abstraction for caption programs is probably going to be `scriptURL`, not `hastebinID`. Then you can have `CaptionProgram.ts` loadf `file://`  URLs from the disk, and `https://` URLs from the web, and the user doesn't need to care if it's Hastebin or something else.

Reply:
I agree, was just keeping it as close to original as possible for our first text-overlay release 👍 then we can improve from there